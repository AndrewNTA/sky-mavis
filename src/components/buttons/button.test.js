import React from 'react';
import renderer from 'react-test-renderer';
import Button, { SECONDARY, FULL, MEDIUM } from './button';

test('Button should render the primary button with small size', () => {
  const component = renderer.create(
    <Button
      text={'ok'}
      onClick={() => {}}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button should render the secondary button with full size', () => {
  const component = renderer.create(
    <Button
      text={'cancel'}
      onClick={() => {}}
      type={SECONDARY}
      size={FULL}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button should render the primary button with medium size', () => {
  const component = renderer.create(
    <Button
      text={'ok'}
      onClick={() => {}}
      size={MEDIUM}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});