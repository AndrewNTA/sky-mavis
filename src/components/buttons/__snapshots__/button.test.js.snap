// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the primary button with medium size 1`] = `
<button
  className="r-primary-btn medium"
  onClick={[Function]}
>
  ok
</button>
`;

exports[`Button should render the primary button with small size 1`] = `
<button
  className="r-primary-btn small"
  onClick={[Function]}
>
  ok
</button>
`;

exports[`Button should render the secondary button with full size 1`] = `
<button
  className="r-secondary-btn full"
  onClick={[Function]}
>
  cancel
</button>
`;
